### ISSUE TITLE
--reuse-results should give a better error message when trying to reuse from the wrong file format

### ISSUE THREAD

**Issue by pavlohrab** [antismash/antismash#338]
**Describe the bug**
I have run an antismash minimal on a genome sequence. In the run I also specified `--sideload antismash/detection/sideloader/test/data/good.json `, however no additional results were created. When I run the same sideloader option with `--reuse-results <path-to-seq>`, the following error arises:
```
File "/home/pavloh/miniconda3/envs/antismash/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```

**Expected behavior**
Added additional annotations to the antismash results

**To Reproduce**
Just run the `--sideload antismash/detection/sideloader/test/data/good.json ` with any analysis. Run the second time with `--reuse-results` option

**System (please complete the following information):**
 - OS: Manjaro Linux 
 - Antismash version: antiSMASH 6.0.0alpha1

**Comment by kblin** [comment]
The sideloading test file only adds annotations to two potential records: `HM219853.1` (which is a nisin A gene cluster) and `nott-HM219853.1`, which is just in the test to make sure that only the right annotations get loaded. If you want to add annotations to one of your records, you need to have the right record ID in your sideloader json file.

Sideloading also works fine with result reuse, but it is a bit pointless to do this. New clusters or regions added in the detection stage will basically invalidate all the previous results.

The reason you're getting the backtrace on result reuse is that you're passing a GenBank file to the JSON parser that is expecting to parse the antiSMASH result JSON file.

So if you were to run things like this:
```sh
antismash antismash/test/integration/data/nisin.gbk   # generate initial antiSMASH results
antismash --reuse-results nisin/nisin.json --sideload antismash/detection/sideloader/test/data/good.json
```
This would initally create antiSMASH results for the class I lanthipeptide nisin,  and then sideload a "Polyketide" subregion and a T1PKS protocluster on top of it.
And again, because the subregions and protoclusters were modified, this would basically discard all of the previously stored results in `nisin/nisin.json`, so there's really no point in doing this.

That said, I guess we need to improve the error message that occurs when people try to `--reuse-results` on something that's not a JSON file.

**Comment by kblin** [comment]
Fixed by 0bc1c66
